### PR TITLE
migrate health check filter to v2 api

### PR DESCRIPTION
**Pull request by rshriram**
*title*: migrate health check filter to v2 api

*Description*: add functionality to populate health check filter directly from protos
*Risk Level*: Low
*Testing*: unit tests, and existing health check filter tests remain untouched.

### CODE REVIEW

**Comment by mattklein123** [comment]
Let's wait for the other PR to merge and then please rebase this on master.

**Comment by mattklein123** [comment]
please squash rebase on origin/master

File: source/common/config/filter_json.cc [hunk 1]
```diff
@@ -266,6 +266,17 @@ void FilterJson::translateFaultFilter(const Json::Object& json_fault,
   }
 }
 
+void FilterJson::translateHealthCheckFilter(
+    const Json::Object& json_health_check,
+    envoy::api::v2::filter::http::HealthCheck& health_check) {
+  json_health_check.validateSchema(Json::Schema::HEALTH_CHECK_HTTP_FILTER_SCHEMA);
+
+  health_check.mutable_pass_through_mode()->set_value(
```
Comment (mattklein123) [inline source/common/config/filter_json.cc]: JSON_UTIL_SET_BOOL

File: source/server/http/health_check.cc [hunk 2]
```diff
@@ -10,30 +10,28 @@
...
+HttpFilterFactoryCb HealthCheckFilterConfig::createHealthCheckFilter(
+    const envoy::api::v2::filter::http::HealthCheck& health_check, const std::string&,
+    FactoryContext& context) {
+  ASSERT(health_check.has_pass_through_mode());
+  ASSERT(!health_check.endpoint().empty());
 
-  bool pass_through_mode = config.getBoolean("pass_through_mode");
-  int64_t cache_time_ms = config.getInteger("cache_time_ms", 0);
-  std::string hc_endpoint = config.getString("endpoint");
+  bool pass_through_mode = health_check.pass_through_mode().value();
```
Comment (mattklein123) [inline source/server/http/health_check.cc]: sorry as long as you are in here can you make these all const?

**Comment by rshriram** [comment]
Oh yeah good point. I am off from today. I can fix these in a follow up PR when I return later next week or if you feel this is critical, let this stay and I will get to it mid next week.

File: source/server/http/health_check.cc (hunk 2 above)
Comment (mattklein123) [reply source/server/http/health_check.cc]: That's fine we can just merge this and can fix in a follow up.

**Pull request merged**